rfcs/0038-cheque/0038-cheque.md

```diff
@@ -0,0 +1,332 @@
+        Data:
+            sudt_amount: 1000 UDT
+
+Outputs:
+    Cheque Cell:
+        Capacity: 165 CKBytes
+        Lock:
+            code_hash: cheque lock
+            args:  <receiver_secp256k1_blake2b_lock_hash[0..20]> <sender_secp256k1_blake2b_lock_hash[0..20]>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 100 UDT
+    SUDT Cell:
+        Capacity: 834.99 CKBytes
+        Lock: <sender_secp256k1_blake2b_lock_script>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 900 UDT
+
+Witnesses
+    <valid signature for sender public key hash>
+```
+
+In this example 0.01 CKBytes is paid as the transaction fee, in production one should calculate the fee based on factors including transaction size, running cycles as well as network status. 0.01 CKByte will be used in all examples as fees for simplicity.
+
+### Claim
+
+#### 1. Claim via receiver signature
+
+```
+Inputs:
+    Cheque Cell:
+        Capacity: 165 CKBytes
+        Lock:
+            code_hash: cheque lock
+            args: <receiver_secp256k1_blake2b_lock_hash[0..20]> <sender_secp256k1_blake2b_lock_hash[0..20]>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 100 UDT
+    SUDT Cell:
+        Capacity: 200 CKBytes
+        Lock: <another_receiver_lock_script>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 200 UDT
+
+ Outputs :
+    SUDT Cell :
+        Capacity: 199.99 CKBytes
+        Lock: <another_receiver_lock_script>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 300 UDT
+    Empty Cell:
+        Capacity: 165 CKBytes
+        Lock: <sender_secp256k1_blake2b_lock_script>
+
+ Witnesses :
+      < valid signature for another public key hash >
+      < valid signature for receiver public key hash >
+```
+
+When a signature is provided and can be validated by the receiver public key hash, and the sum of sender output cells capacity is equal to the sum of the cheque input cells capacity, the cheque cells can be unlocked. In this example a cheque cell is converted back to a sender empty cell and the SUDT is transferred from the sender to an arbitrary lock script set by the receiver.
+
+#### 2. Claim via receiver lock script
+
+```
+Inputs:
+    Cheque Cell:
+        Capacity: 165 CKBytes
+        Lock:
+            code_hash: cheque lock
+            args: <receiver_secp256k1_blake2b_lock_hash[0..20]> <sender_secp256k1_blake2b_lock_hash[0..20]>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 100 UDT
+    SUDT Cell:
+        Capacity: 200 CKBytes
+        Lock: <receiver_secp256k1_blake2b_lock_script>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 200 UDT
+
+Outputs:
+    SUDT Cell:
+        Capacity: 199.99 CKBytes
+        Lock: <receiver_secp256k1_blake2b_lock_script>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 300 UDT
+    Empty Cell:
```

Comment:
I think this not necessary be `Empty Cell`, use `Any Cell` would be better?

Reply:
@janx any suggestions?

Reply:
I think `Empty Cell` is appropriate for this example because that's what is demonstrated. `Any Cell` sounds good too. Maybe it's even better to add a comment to make clear that any cell can be used here.